### ISSUE TITLE
cds-icon-button component gives ariaLabel required warning even though its provided

### ISSUE THREAD

**Issue by tushargupta1301** [vmware/clarity#6463]
## Describe the bug

<CdsIconButton
        action={action}
        ariaLabel={iconTitle}

On providing ariaLabel in CdsIconButton, it gets converted to arialabel on the html. This ends up with a lot of warnings in the console.
ariaLabel should be converted to aria-label on html.

~Link to slack conversation: EDIT: link removed~

## How to reproduce

Use CdsIconButton with ariaLabel.
Check inspect element console.

## Expected behavior

Console should be clear of warnings
## Versions

**Clarity project:**

- [ ^5.5.4] Clarity Core

**Clarity version:**

- [^5.5.4 ] v5.x

**Framework:**

- [^5.5.4 ] React

**Framework version:**
_ie: React ^16.8.6

**Device:**

- Type: [MacBook]
- OS: [iOS]
- Browser [chrome]

**Comment by gnomeontherun** [comment]
This appears to be linked to a VMware product, so can @scroniser confirm we have this internally?

**Comment by ashleyryan** [comment]
The issue here is that unlike every other attribute, aria attributes are named with a hyphen. So even suppling `ariaLabel` won't work, and the message will still appear. It needs to be `aria-label`. I think we can tweak our messaging to better handle this.

Core put together an example and some links:

- https://stackblitz.com/edit/http-server-1s23vj?file=src%2FApp.tsx
- https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaLabel
- https://github.com/facebook/react/issues/17378
- https://reactjs.org/docs/dom-elements.html

**Comment by scroniser** [comment]
Sorry, I didn't see this before vacation. @ashleyryan is correct, aria-label must use a hyphen to be a valid attribute. I don't believe we had it tracked internally (I can explain why if folks would like to raise it at our next a11y<>Clarity meeting).
